src/pcre2_compile.c

```diff
@@ -6520,16 +6520,18 @@ for (;; pptr++)
     case META_COND_RNUMBER:   /* (?(Rdigits) */
     case META_COND_NAME:      /* (?(name) or (?'name') or ?(<name>) */
     case META_COND_RNAME:     /* (?(R&name) - test for recursion */
-    case META_SCS_NAME:       /* Name of scan substring */
+    case META_CAPTURE_NAME:   /* Generic capture name */
     bravalue = OP_COND;
+
+    if (lengthptr != NULL)
       {
-      int count, index;
-      unsigned int i;
+      uint32_t i;
       PCRE2_SPTR name;
       named_group *ng = cb->named_groups;
+      uint32_t *start_pptr = pptr;
       uint32_t length = *(++pptr);
 
-      if (meta == META_SCS_NAME)
+      if (meta == META_CAPTURE_NAME)
```

Comment:
You renamed META_SCS_NAME to META_CAPTURE_NAME, which is fine. But as far as I can tell, it's still being used only by scan_substring? Just checking I haven't missed anything. I can't see any new places where it's generated. In future of course we could re-use now that it has a more generic name.

Reply:
Currently only scs uses it. I have plans for using it for other cases in the future.